### ISSUE TITLE
[1.16 regression] Relative order of property methods changes mypy output

### ISSUE THREAD

**Issue by gpauloski** [python/mypy#19224]
**Bug Report**

In mypy 1.16, the relative ordering of property methods (i.e., `getter`/`setter`/`deleter`) can produce an "Invalid property setter signature" error.

**To Reproduce**

[mypy playground with the same example](https://mypy-play.net/?mypy=1.16.0&python=3.13&gist=18c775f7322ed5b9011445f36172dcbf)

Having an `@deleter` or `@getter` before a `@setter` property method produces an error:
```python
class C:
    _value: int = 0

    @property
    def value(self) -> int:
        return self._value

    @value.deleter
    def value(self) -> None: ...  # error: Invalid property setter signature  [misc]

    @value.setter
    def value(self, v: int) -> None: ...
```
```
$ mypy t.py
main.py:9: error: Invalid property setter signature  [misc]
Found 1 error in 1 file (checked 1 source file)
```

Move the `@value.deleter` method to after `@value.setter` and the error is fixed.
```python
class C:
    _value: int = 0

    @property
    def value(self) -> int:
        return self._value

    @value.setter
    def value(self, v: int) -> None: ...

    @value.deleter
    def value(self) -> None: ...
```
```
$ mypy t.py
Success: no issues found in 1 source file
```

**Expected Behavior**

The relative ordering should not alter the output of mypy. I'm not aware of any reason why the ordering should matter and all orderings pass mypy in 1.15.

**Your Environment**

- Mypy version used: 1.16
- Mypy command-line flags: None
- Mypy configuration options from `mypy.ini` (and other config files): None
- Python version used: 3.13

**Comment by sterliakov** [comment]
Introduced in #18510, TODO added in #18774

https://github.com/python/mypy/blob/4fe7d9f8e2b47d95b39426a4ea5c4b39e4b229c6/mypy/checker.py#L677-L683

This is a known limitation for now; cc @ilevkivskyi

**Comment by ilevkivskyi** [comment]
Yeah, I was too tired to handle that, and TBH still tired. This looks lower priority than other things (unless this causes dozens of errors for you).
